cmd/ddev/cmd/utility-mutagen-diagnose.go

```diff
@@ -0,0 +1,407 @@
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/ddev/ddev/pkg/version"
+	"github.com/ddev/ddev/pkg/versionconstants"
+	"github.com/spf13/cobra"
+)
+
+// MutagenDiagnoseCmd implements the ddev utility mutagen-diagnose command
+var MutagenDiagnoseCmd = &cobra.Command{
+	Use:   "mutagen-diagnose",
+	Short: "Diagnose Mutagen sync configuration and performance",
+	Long: `Analyze Mutagen sync status, volume sizes, and configuration issues.
+
+This command checks:
+- Volume sizes and disk usage
+- Upload_dirs configuration
+- Sync session status and problems
+- Performance issues (node_modules, large files, etc.)
+
+Use the --all flag to analyze all Mutagen volumes system-wide.`,
+	Example: `ddev utility mutagen-diagnose
+ddev ut mutagen-diagnose
+ddev utility mutagen-diagnose --all  # Show all Mutagen volumes`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 0 {
+			util.Failed("This command takes no additional arguments")
+		}
+
+		showAll, _ := cmd.Flags().GetBool("all")
+		exitCode := runMutagenDiagnose(showAll)
+		os.Exit(exitCode)
+	},
+}
+
+func init() {
+	MutagenDiagnoseCmd.Flags().BoolP("all", "a", false, "Show all Mutagen volumes system-wide")
+	DebugCmd.AddCommand(MutagenDiagnoseCmd)
+}
+
+// runMutagenDiagnose performs the diagnostic checks and outputs results
+// Returns exit code: 0 if no issues, 1 if issues found
+func runMutagenDiagnose(showAll bool) int {
+	var app *ddevapp.DdevApp
+	hasIssues := false
+
+	// Try to load app from current directory (optional)
+	app, _ = ddevapp.GetActiveApp("")
+
+	// If showAll is true or we're not in a project, show system-wide analysis
+	if showAll || app == nil {
+		hasIssues = showAllMutagenVolumes()
+		if app == nil {
+			// Not in a project directory and not using --all
+			if !showAll {
+				output.UserOut.Println("Not in a DDEV project directory.")
+				output.UserOut.Println("Use --all flag to analyze all Mutagen volumes system-wide.")
+				output.UserOut.Println()
+			}
+			if hasIssues {
+				return 1
+			}
+			return 0
+		}
+	}
+
+	// If we have an app, analyze it
+	if app != nil {
+		// Check if Mutagen is enabled for this project
+		if !app.IsMutagenEnabled() {
+			output.UserOut.Printf("Mutagen is not enabled for project '%s'\n", app.Name)
+			output.UserOut.Println()
+			output.UserOut.Println("To enable Mutagen:")
+			output.UserOut.Println("  ddev config --performance-mode=mutagen")
+			output.UserOut.Println("  ddev restart")
+			return 0
+		}
+
+		// Check if project is running - start it if needed
+		status, _ := app.SiteStatus()
+		if status != ddevapp.SiteRunning {
+			output.UserOut.Printf("Project '%s' is not running. Starting project to enable sync diagnostics...\n", app.Name)
+			output.UserOut.Println()
+			err := app.Start()
+			if err != nil {
+				util.Failed("Failed to start project: %v", err)
+				return 1
+			}
+		}
+
+		projectHasIssues := showProjectDiagnostics(app)
+		if projectHasIssues {
+			hasIssues = true
+		}
+	}
```

Comment:
app is not nil here, app=nil completely handled above:

```suggestion
	// Check if Mutagen is enabled for this project
	if !app.IsMutagenEnabled() {
		output.UserOut.Printf("Mutagen is not enabled for project '%s'\n", app.Name)
		output.UserOut.Println()
		output.UserOut.Println("To enable Mutagen:")
		output.UserOut.Println("  ddev config --performance-mode=mutagen")
		output.UserOut.Println("  ddev restart")
		return 0
	}

	// Check if project is running - start it if needed
	status, _ := app.SiteStatus()
	if status != ddevapp.SiteRunning {
		output.UserOut.Printf("Project '%s' is not running. Starting project to enable sync diagnostics...\n", app.Name)
		output.UserOut.Println()
		err := app.Start()
		if err != nil {
			util.Failed("Failed to start project: %v", err)
			return 1
		}
	}

	projectHasIssues := showProjectDiagnostics(app)
	if projectHasIssues {
		hasIssues = true
	}
```